src/Classes/PolygonMask.ts

```diff
@@ -0,0 +1,222 @@
+        };
+
+        for (let i = 0; i < len; i++) {
+            const pathSeg: any = this.path.pathSegList.getItem(i);
+
+            switch (pathSeg.pathSegType) {
+                //
+                // Absolute
+                //
+                case window.SVGPathSeg.PATHSEG_MOVETO_ABS:
+                case window.SVGPathSeg.PATHSEG_LINETO_ABS:
+                case window.SVGPathSeg.PATHSEG_CURVETO_CUBIC_ABS:
+                case window.SVGPathSeg.PATHSEG_CURVETO_QUADRATIC_ABS:
+                case window.SVGPathSeg.PATHSEG_ARC_ABS:
+                case window.SVGPathSeg.PATHSEG_CURVETO_CUBIC_SMOOTH_ABS:
+                case window.SVGPathSeg.PATHSEG_CURVETO_QUADRATIC_SMOOTH_ABS:
+                    p.x = pathSeg.x;
+                    p.y = pathSeg.y;
+
+                case window.SVGPathSeg.PATHSEG_LINETO_HORIZONTAL_ABS:
+                    p.x = pathSeg.x;
+                    break;
+
+                case window.SVGPathSeg.PATHSEG_LINETO_VERTICAL_ABS:
+                    p.y = pathSeg.y;
+                    break;
+
+                //
+                // Relative
+                //
+                case window.SVGPathSeg.PATHSEG_LINETO_REL:
+                case window.SVGPathSeg.PATHSEG_MOVETO_REL:
+                case window.SVGPathSeg.PATHSEG_CURVETO_CUBIC_REL:
+                case window.SVGPathSeg.PATHSEG_CURVETO_QUADRATIC_REL:
+                case window.SVGPathSeg.PATHSEG_ARC_REL:
+                case window.SVGPathSeg.PATHSEG_CURVETO_CUBIC_SMOOTH_REL:
+                case window.SVGPathSeg.PATHSEG_CURVETO_QUADRATIC_SMOOTH_REL:
+                    p.x += pathSeg.x;
+                    p.y += pathSeg.y;
+                    break;
+
+                case window.SVGPathSeg.PATHSEG_LINETO_HORIZONTAL_REL:
+                    p.x += pathSeg.x;
+                    break;
+                case window.SVGPathSeg.PATHSEG_LINETO_VERTICAL_REL:
+                    p.y += pathSeg.y;
+                    break;
+
+                case window.SVGPathSeg.PATHSEG_UNKNOWN:
+                case window.SVGPathSeg.PATHSEG_CLOSEPATH:
+                    continue; // Skip the closing path (and the UNKNOWN)
+                    break;
+            }
+
+            polygonRaw.push([p.x + this.offset.x, p.y + this.offset.y]);
+        }
+
+        return polygonRaw;
+    }
+
+    drawDebugPolygon() {
+        const container = this.container;
+        const options = container.options;
+        const context = container.canvas.context;
+
+        if (context && this.raw) {
+            context.beginPath();
+            context.moveTo(this.raw[0][0], this.raw[0][1]);
+
+            for (let i = 1; i < this.raw.length; i++) {
+                context.lineTo(this.raw[i][0], this.raw[i][1]);
+            }
+
+            context.closePath();
+            context.strokeStyle = options.polygon.debug.color;
+            context.lineWidth = 0.5;
+            context.stroke();
+        }
+    }
+
+    drawPointsOnPolygonPath() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [expected call-signature: 'drawPointsOnPolygonPath' to have a typedef](https://app.codacy.com/manual/ar3s/tsparticles/pullRequest?prid=5042590)